lib/widget/triple_state_switch.dart

```diff
@@ -0,0 +1,296 @@
+import 'package:ensemble/framework/event.dart';
+import 'package:ensemble/framework/extensions.dart';
+import 'package:ensemble/screen_controller.dart';
+import 'package:ensemble/util/utils.dart';
+import 'package:ensemble/widget/helpers/widgets.dart';
+import 'package:ensemble/widget/input/form_helper.dart';
+import 'package:ensemble/framework/action.dart' as framework;
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+import 'package:flutter/material.dart';
+
+class EnsembleTripleStateSwitch extends StatefulWidget
+    with
+        Invokable,
+        HasController<TripleStateSwitchController,
+            EnsembleTripleStateSwitchState> {
+  static const type = 'TripleStateSwitch';
+  EnsembleTripleStateSwitch({Key? key}) : super(key: key);
+
+  final TripleStateSwitchController _controller = TripleStateSwitchController();
+  @override
+  TripleStateSwitchController get controller => _controller;
+
+  @override
+  State<StatefulWidget> createState() => EnsembleTripleStateSwitchState();
+
+  @override
+  Map<String, Function> getters() {
+    return {'value': () => _controller.value.name};
+  }
+
+  @override
+  Map<String, Function> methods() {
+    return {};
+  }
+
+  @override
+  Map<String, Function> setters() {
```

Comment:
why copy and pasted all these code? 99% of it identical to the regular switch. You should at least use a base class instead of replicating all these.

Reply:
Refactored